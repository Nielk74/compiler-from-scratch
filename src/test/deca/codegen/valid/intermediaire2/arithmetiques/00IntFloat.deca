// Description:
//    Programme affichant les résultats d'opérations arithmétiques entre des variables int et float
//
// Resultats:
//    00IntFloat.expected
//
// Historique:
//    cree le 08/06/2022

{
    int x = 5;
    int z = 1;
    float y = 0.5;
    float a = 5.9;

    // opérations entre int et float
    println(x + y);
    println(x - y);
    println(x * y);
    println(x / y);

    println(y + x);
    println(y - x);
    println(y * x);
    println(y / x);

    // opérations entre ints
    println(x + z);
    println(x - z);
    println(x * z);
    println(x / z);

    println(x + 5);
    println(x - 35);
    println(x * 156);
    println(x / 9);

    println(5 + x);
    println(35 - x);
    println(156 * x);
    println(9 / x);    

    // opérations entre floats
    println(y + a);
    println(y - a);
    println(y * a);
    println(y / a);

    println(y + 5.6);
    println(y - 560.05);
    println(y * 98.1654);
    println(y / 5465.17);

    println(5.6 + a);
    println(560.05 - a);
    println(98.1654 * a);
    println(5465.17 / a);

    // tests modulo
    x = 5;
    println(x % 3);
    println(24 % x);
    z = 3;
    println(x % z);
}