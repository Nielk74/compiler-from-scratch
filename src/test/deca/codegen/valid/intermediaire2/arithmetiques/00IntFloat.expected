5.50000e+00
4.50000e+00
2.50000e+00
1.00000e+01
5.50000e+00
-4.50000e+00
2.50000e+00
1.00000e-01
6
4
5
5.00000e+00
10
-30
780
5.55556e-01
10
30
780
1.80000e+00
6.40000e+00
-5.40000e+00
2.95000e+00
8.47458e-02
6.10000e+00
-5.59550e+02
4.90827e+01
9.14885e-05
1.15000e+01
5.54150e+02
5.79176e+02
9.26300e+02
2
4
2
